src/utils/url.js

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+const defaultUrl = {
+  hash: "",
+  host: "",
+  hostname: "",
+  href: "",
+  origin: "null",
+  password: "",
+  path: "",
+  pathname: "",
+  port: "",
+  protocol: "",
+  search: "",
+  searchParams: {}, // This should be a "URLSearchParams" object
+  username: ""
+};
+
+export function parse(url: string): URL | object {
+  try {
+    const urlObj = new URL(url);
+    urlObj.path = urlObj.pathname + urlObj.search;
```

Comment:
"path" isn't a native URL property -- I found this in URL's readme

Reply:
nice find